lib/dynamoid/transaction_write/save.rb

```diff
@@ -86,6 +92,30 @@ def action_request
         end
       end
 
+      # sets a value in the attributes
+      def set(attributes)
+        @model.assign_attributes(attributes)
+      end
+
+      # adds to array of fields for use in REMOVE update expression
+      def remove(field)
+        @removals << field
+      end
+
+      # increments a number or adds to a set, starts at 0 or [] if it doesn't yet exist
+      def add(values)
+        @additions.merge!(values)
+      end
+
+      # deletes a value or values from a set type or simply sets a field to nil
+      def delete(field_or_values)
+        if field_or_values.is_a?(Hash)
+          @deletions.merge!(field_or_values)
+        else
+          remove(field_or_values)
+        end
+      end
```

Comment:
It makes sense to have a separate class to collect user actions (I mean to set/add/delete), similar to `ItemUpdater` (https://github.com/Dynamoid/dynamoid/blob/master/lib/dynamoid/adapter_plugin/aws_sdk_v3/item_updater.rb)